apollo-server/v3/integration-testing/server.ts

```diff
@@ -0,0 +1,46 @@
+// @ts-check
+const { createServer } = require('http');
+const express = require('express');
+export const { ApolloServer, gql } = require('apollo-server-express');
+
+// Schema definition
+export const typeDefs = gql`
+  type Query {
+    hello(name: String): String!
+  }
+`;
+
+// Resolver map
+export const resolvers = {
+  Query: {
+    hello: (_, { name }) => `Hello ${name}!`,
+  },
+};
+
+(async () => {
```

Comment:
I suggest having a function in this file that creates a server (and perhaps awaits listen, with an argument passed by the caller?) and then a second file `main.ts` that runs it at the top level, instead of deciding what to do based on env vars and duplicating the AS setup code across files.

In an ideal world you could use top-level await for the `main.ts` file, though I've found it challenging to get that working with `ts-node`. If you look at my recent example you'll see I get top-level await working but only by doing a manual `tsc && node` thing, and I needed to set various parameters in package.json and tsconfig.json to get that working.

Reply:
Love this suggestion so much DRYer! I created a `main.ts` which creates the server, and now `server.ts` just exports a function that can be called to create a server. 

I couldn't get a top-level `await` to work without typescript yelling at me about the jest types, so I ended up nesting creating the server within an async function in `main.ts`.